internal/exec/terraform_state_utils.go

```diff
@@ -58,8 +58,21 @@ func GetTerraformState(
 		}
 	}
 
+	message := fmt.Sprintf("Fetching %s output from %s in %s", output, component, stack)
+
+	if atmosConfig != nil && (atmosConfig.Logs.Level == u.LogLevelTrace || atmosConfig.Logs.Level == u.LogLevelDebug) {
+		// Initialize spinner.
+		p := NewSpinner(message)
+		spinnerDone := make(chan struct{})
+		// Run spinner in a goroutine.
+		RunSpinner(p, spinnerDone, message)
+		// Ensure the spinner is stopped before returning.
+		defer StopSpinner(p, spinnerDone)
+	}
```

Comment:
Log levels are not feature flags for other functionality. Something is wrong here. Generally no logic should be tied to log levels. Log Levels != TUI. Spinners are TUI.